### PR TITLE
ENH: Improve short reprs

### PR DESCRIPTION
**Pull request by jnazaren**
Improves the representation of some nodes in the pipeline graph

### CODE REVIEW

File: zipline/pipeline/expression.py [hunk 1]
```diff
@@ -328,7 +330,12 @@ def __repr__(self):
 
     def short_repr(self):
         """Short repr to use when rendering Pipeline graphs."""
-        return "Expression: {expr}".format(
+        final = self._expr
+        num_list = re.findall(r"[-+]?\d*\.\d+", self._expr)
```
Comment (ssanderson) [inline zipline/pipeline/expression.py]: Is the purpose of this to replace numbers with scientific notation? If so, do we need to use `Decimal` here (vs. say, `float`)?

File: zipline/pipeline/factors/factor.py [hunk 2]
```diff
@@ -1411,6 +1411,13 @@ def __repr__(self):
             mask=self.mask,
         )
 
+    def short_repr(self):
+        return "Rank({})\nmethod={}\nmask={}".format(
```
Comment (ssanderson) [inline zipline/pipeline/factors/factor.py]: It feels a little odd to me to have some of the parameters to `Rank` be inside parentheses, but to have method and mask outside. What do you think about something more yaml-ish like this?

```
Rank:
  method: 'ordinal'
  mask: SomeFilter 
```

File: zipline/pipeline/filters/filter.py [hunk 3]
```diff
@@ -481,6 +488,12 @@ def _compute(self, arrays, dates, assets, mask):
         data = arrays[0]
         return params['op'](data, *params['opargs']) & mask
 
+    def short_repr(self):
+        return "{}\nop={}.{}()".format(type(self).__name__,
```
Comment (ssanderson) [inline zipline/pipeline/filters/filter.py]: flake8 is going to complain about the formatting here. I would format this as

```python
"...".format(
    type(self).__name__,
    ...,
    ...,
)
```

File: zipline/pipeline/expression.py [hunk 4]
```diff
@@ -328,7 +330,12 @@ def __repr__(self):
...
+        for i in num_list:
+            final = final.replace(i, '%.2E' % Decimal(i))
```
Comment (richafrank) [inline zipline/pipeline/expression.py]: How does the loop compare to `re.sub`? I could see it being potentially more efficient, but I don't know if it matters here.

Comment (jnazaren) [reply zipline/pipeline/expression.py]: Oh I'll use that instead - str.replace() is an old habit I have from Java

Comment (llllllllll) [inline zipline/pipeline/expression.py]: small tip: you can get access to the `%` formatting behavior for a single object with the `format` builtin:

```python
In [6]: format(Decimal(1), '.2E')
Out[6]: '1.00E+0'
```

File: zipline/pipeline/expression.py (hunk 1 above)
Comment (jnazaren) [reply zipline/pipeline/expression.py]: No, that was just from the original example of shortening numbers I found - I've changed it to `float`

File: zipline/pipeline/factors/factor.py (hunk 2 above)
Comment (jnazaren) [reply zipline/pipeline/factors/factor.py]: Ok, yeah that makes sense

File: zipline/pipeline/expression.py [hunk 5]
```diff
@@ -328,7 +326,9 @@ def __repr__(self):
 
     def short_repr(self):
         """Short repr to use when rendering Pipeline graphs."""
-        return "Expression: {expr}".format(
-            typename=type(self).__name__,
-            expr=self._expr,
+        final = re.sub(r"[-+]?\d*\.\d+",
```
Comment (ssanderson) [inline zipline/pipeline/expression.py]: Can we add a comment explaining what's happening here?

File: zipline/pipeline/expression.py [hunk 6]
```diff
@@ -328,7 +326,9 @@ def __repr__(self):
...
+                       lambda x: format(float(x.group(0)), '.2E'),
+                       self._expr)
+        return "Expression:\l{}\l".format(
```
Comment (ssanderson) [inline zipline/pipeline/expression.py]: In our other uses of `\l` it looks like we're putting two leading spaces after the line break. Any reason not to do that here too"

Comment (jnazaren) [reply zipline/pipeline/expression.py]: Sure, I can add that in

File: zipline/pipeline/expression.py (hunk 5 above)
Comment (jnazaren) [reply zipline/pipeline/expression.py]: Yep

**Comment by ssanderson** [comment]
LGTM, feel free to merge when ready.

**Comment by richafrank** [comment]
Nice! Before merging, would be good to clean up the WIP commit history.

**Pull request merged**